packages/slate-editor/src/modules/components/BookmarkCard/Details.tsx

```diff
@@ -1,19 +1,53 @@
+import classNames from 'classnames';
 import type { PropsWithChildren } from 'react';
+import { useMemo } from 'react';
 import React from 'react';
 
 import { utils, stripTags } from '#lib';
 
 import styles from './BookmarkCard.module.scss';
 
 interface DetailsProps {
+    hasThumbnail: boolean;
+    layout: 'vertical' | 'horizontal';
     title?: string;
     description?: string;
     href: string;
 }
 
-export function Details({ title, description, href, children }: PropsWithChildren<DetailsProps>) {
+export function Details({
+    title,
+    description,
+    href,
+    children,
+    layout,
+    hasThumbnail,
+}: PropsWithChildren<DetailsProps>) {
+    const { isDescriptionShort, isTitleShort } = useMemo(() => {
+        let titleOneLineLength;
+        let descriptionOneLineLength;
+
+        if (layout === 'vertical' || !hasThumbnail) {
+            titleOneLineLength = 50;
+            descriptionOneLineLength = 75;
+        } else {
+            titleOneLineLength = 28;
+            descriptionOneLineLength = 45;
+        }
+
+        return {
+            isTitleShort: (title ?? '').length <= titleOneLineLength,
+            isDescriptionShort: (description ?? '')?.length <= descriptionOneLineLength,
+        };
+    }, [title, description, layout, hasThumbnail]);
```

Comment:
IDK, I know this is based on what the designers requested. But I think this is unnecessarily complex. I'd not expect web-presentation JS counting characters. 

But alright. :man_shrugging: